### DISCUSSION TITLE
riverpod_generator The argument type 'InputDpNotifier Function(DPRepository)' can't be assigned to the parameter type 'InputDpNotifier Function()'.

### DISCUSSION THREAD

**Discussion by ghifariakbar01** [rrousselGit/riverpod#2989] [Q&A]
it says 'InputDpNotifier' is from 'package:flutter_project_template_riverpod/features/home/application/dp/input_dp_notifier.dart' ('lib/features/home/application/dp/input_dp_notifier.dart').

the following are code snippet for InputDpNotifier

```
@riverpod
class InputDpNotifier extends _$InputDpNotifier {
  final DPRepository _repository;

  InputDpNotifier(DPRepository repository) : _repository = repository;

  @override
  InputDPState build() => InputDPState.initial();

  void changeNoMR(String noMrStr) {
    state = state.copyWith(noMR: NoMR(noMrStr));
  }
 
 ...
}
```

and DPRepository which is a InputDpNotifier constructor

```

class DPRepository {
  DPRepository(this._remoteService);

  final DPRemoteService _remoteService;

  Future<Either<RemoteFailure, Dp>> getDPByCode(String paymentCode) async {
    try {
      final assets = await _remoteService.getDPByCode(paymentCode);

      return right(assets);
    } on RestApiException catch (error) {
      return left(RemoteFailure.server(error.errorCode, error.message));
    } on FormatException catch (error) {
      return left(RemoteFailure.parse(message: 'Parsing Error: $error'));
    } on NoConnectionException {
      return left(RemoteFailure.noConnection());
    }
  }

....

}
```

the lints doesn't give out errors but during run time it says The argument type 'InputDpNotifier Function(DPRepository)' can't be assigned to the parameter type 'InputDpNotifier Function()'. 

- 'InputDpNotifier' is from 'package:flutter_project_template_riverpod/features/home/application/dp/input_dp_notifier.dart' ('lib/features/home/application/dp/input_dp_notifier.dart').
input_dp_notifier.dart:1

- 'DPRepository' is from 'package:flutter_project_template_riverpod/features/home/infrastructure/dp/dp_repository.dart' ('lib/features/home/infrastructure/dp/dp_repository.dart').
dp_repository.dart:1

  InputDpNotifier.new,
  ^

Target kernel_snapshot failed: Exception

Help ?

**Comment by rrousselGit** [comment]
You're not allowed to specify a constructor with parameters.

**Reply by ghifariakbar01** [reply]
How would you approach the situation? For DpRepository to used in Notifier?
